__tests__/__unit__/extension.unit.test.ts

```diff
@@ -803,35 +871,40 @@ describe("Extension Unit Tests", () => {
         });
 
         showInformationMessage.mockReset();
-        (profileLoader.loadAllProfiles as any).mockReset();
-        (profileLoader.loadAllProfiles as any).mockReturnValueOnce([]);
-
-        await extension.addJobsSession(testJobsTree);
-
-        expect(showInformationMessage.mock.calls.length).toBe(1);
-        expect(showInformationMessage.mock.calls[0][0]).toEqual("No more profiles to add");
-
-        showInformationMessage.mockReset();
-        (profileLoader.loadAllProfiles as any).mockReset();
-        (profileLoader.loadAllProfiles as any).mockReturnValueOnce([]);
+        Object.defineProperty(profileLoader.Profiles, "getInstance", {
+            value: jest.fn(() => {
+                return {
+                    allProfiles: [],
+                    defaultProfile: undefined
+                };
+            })
+        });
         await extension.addJobsSession(testJobsTree);
 
         expect(showInformationMessage.mock.calls.length).toBe(1);
         expect(showInformationMessage.mock.calls[0][0]).toEqual("No more profiles to add");
+        //  TODO these should be tests on the profileLoader file
```

Comment:
Thanks for adding a not as to why these tests are commented out. :+1:

Reply:
Yes I will be reworking the profile class and try and get the tests all in the correct place